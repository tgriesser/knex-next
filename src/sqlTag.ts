import { RawNode } from "./data/datatypes";
import { List } from "immutable";

export function sql(query: TemplateStringsArray, ...bindings: any[]) {
  return RawNode({
    fragments: List(query),
    bindings: List(bindings),
  });
}

export function ident() {}
