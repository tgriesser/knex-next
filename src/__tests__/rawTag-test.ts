import { sql } from "../sqlTag";

test("raw tag", () => {
  expect(sql`some value`).toMatchSnapshot();
});

test("raw tag with bindings", () => {
  expect(sql`select * from tbl where id = ${1}`).toMatchSnapshot();
});
