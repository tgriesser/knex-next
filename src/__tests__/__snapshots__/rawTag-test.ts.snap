// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`raw tag 1`] = `
Immutable.RawNode {
  "__typename": "RawNode",
  "fragments": Immutable.List [
    "some value",
  ],
  "bindings": Immutable.List [],
}
`;

exports[`raw tag with bindings 1`] = `
Immutable.RawNode {
  "__typename": "RawNode",
  "fragments": Immutable.List [
    "select * from tbl where id = ",
    "",
  ],
  "bindings": Immutable.List [
    1,
  ],
}
`;
